src/main/java/org/tron/common/runtime/Runtime.java

```diff
@@ -518,42 +544,58 @@ public void go() {
         }
       }
     } catch (OutOfResourceException e) {
-      spendUsage(0);
       logger.error(e.getMessage());
       runtimeError = e.getMessage();
+      throw new OutOfSlotTimeException(e.getMessage());
     } catch (Exception e) {
       logger.error(e.getMessage());
       runtimeError = e.getMessage();
     }
   }
 
-  private void spendUsage(long useedStorageSize) {
+  private boolean spendUsage(long useedStorageSize) {
 
-    cpuProcessor = new CpuProcessor(deposit.getDbManager());
+    long cpuUsage = result.getGasUsed();
 
-    long now = System.nanoTime() / 1000;
-    long cpuUsage = now - program.getVmStartInUs();
+    ContractCapsule contract = deposit.getContract(result.getContractAddress());
+    ByteString originAddress = contract.getInstance().getOriginAddress();
+    AccountCapsule origin = deposit.getAccount(originAddress.toByteArray());
+    long originResourcePercent = 100 - contract.getConsumeUserResourcePercent();
+    originResourcePercent = min(originResourcePercent, 100);
+    originResourcePercent = max(originResourcePercent, 0);
+    long originCpuUsage = cpuUsage * originResourcePercent / 100;
+    originCpuUsage = min(originCpuUsage, cpuProcessor.getAccountLeftCpuInUsFromFreeze(origin));
+    long callerCpuUsage = cpuUsage - originCpuUsage;
 
-//    ContractCapsule contract = deposit.getContract(result.getContractAddress());
-//    ByteString originAddress = contract.getInstance().getOriginAddress();
-//    AccountCapsule origin = deposit.getAccount(originAddress.toByteArray());
     if (useedStorageSize <= 0) {
-      trace.setBill(cpuUsage, 0);
-      return;
+      trace.setBill(callerCpuUsage, 0);
+      return true;
     }
+    long originStorageUsage = useedStorageSize * originResourcePercent / 100;
+    originStorageUsage = min(originCpuUsage, origin.getStorageLeft());
+    long callerStorageUsage = originStorageUsage - originStorageUsage;
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Correct one of the identical sub-expressions on both sides of operator "-" ([squid:S1764](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1764))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1260&issues=AWUSgvRw8anAXINB2Cnh&open=AWUSgvRw8anAXINB2Cnh)

[ISSUE_KEY]: AWUSgvRw8anAXINB2Cnh